File: LeetCode/majorityElement.js
const majorityElement = (nums) => {
  const ct = nums.length / 2;
  const map = {};
  for (const num of nums) {
    if (!map[num]) {
      map[num] = 1;
    } else {
      map[num] += 1;
    }
    if (map[num] > ct) {
      return num;
    }
  }
};

File: LeetCode/DailyDoseCSharp/MajorityElement.cs
public class Solution {
    public int MajorityElement(int[] nums) {
        var numsDictionary = new Dictionary<int, int>();
        int appearance = nums.Length / 2;
        foreach (int num in nums)
        {
            if (numsDictionary.ContainsKey(num))
            {
                numsDictionary[num]++;
            }
            else
            {
                numsDictionary.Add(num, 1);
            }
            if (numsDictionary[num] > appearance)
            {
                return num;
            }
        }
        return -1;
    }
}
